src/app/server/AclStorage.cpp

```diff
@@ -0,0 +1,488 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/server/AclStorage.h>
+
+#include <app/server/Server.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::Access;
+
+using EncodableEntry   = AclStorage::EncodableEntry;
+using Entry            = AccessControl::Entry;
+using EntryListener    = AccessControl::EntryListener;
+using StagingAuthMode  = Clusters::AccessControl::AuthMode;
+using StagingPrivilege = Clusters::AccessControl::Privilege;
+using StagingTarget    = Clusters::AccessControl::Structs::Target::Type;
+using Target           = AccessControl::Entry::Target;
+
+namespace {
+
+/*
+Size calculation for TLV encoded entry.
```

Comment:
Is there a reason not to use `TLV::EstimateStructOverhead` here?  I guess it might be a pain for the lists... Worth at least a comment about why it's not being used.

Reply:
I'm still going to have to come back and revisit that last TODO (in another PR, I'd like to get this one in) so I'll add any relevant comments about these functions when I do that (I made a note of it).